src/ui/component/notification/notification.tsx

```diff
@@ -0,0 +1,54 @@
+import type { FC } from 'react'
+import { useMemo, useCallback } from 'react'
+import { useAppStore } from '@/ui/store/appStore'
+import type { IconName } from '@/ui/component/icon/icon'
+import { Toast } from '@/ui/component/toast/toast'
+import type { NotificationType } from '@/domain/notification/entity'
+import { toEffectfulObject } from '@/util/effect'
+import { useNotificationStore } from '@/ui/store'
+import type { DismissNotificationInput } from '@/domain/notification/aggregate'
+
+export const Notification: FC = () => {
+  const { notifications, dismissNotification } = toEffectfulObject(
+    useNotificationStore(state => ({
+      notifications: state.notifications,
+      dismissNotification: state.dismissNotification
+    }))
+  )
+  const theme = useAppStore(store => store.theme)
+  const notificationData = useMemo(() => notifications(), [notifications])
```

Comment:
Do we need to utilize a memo here? Considering that the notifications are backed by the zustand store, it seems unnecessary, imho. I suggest removing it and directly calling the function at the location where it's needed.

Reply:
You're right. Zustand helps to avoid unnecessary re-renders by subscribing to the specific parts of the state that a component is using. In this case memoization with `useMemo` is not necessary because the state is already being efficiently managed by Zustand, and any updates to the notifications will trigger a re-render only when required.